Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -242,6 +247,18 @@ public class DocumentIterator: IteratorProtocol {
         return values
     }
 
+    /// Returns the current value (equivalent to the `currentValue` property) or throws on error.
+    internal func safeCurrentValue() throws -> BSONValue? {
+        switch self.currentType {
+        case .symbol:
```

Comment:
I realize the original `currentValue` property accessor had the same logic, but does this mean that the Swift driver cannot round-trip Symbol and DBPointer types w/o converting them to strings and documents? Or perhaps they can be roundtripped so long as we leave them in their Document and don't explicitly get them and set them on a new document to be stored.

Reply:
yup, all of that is correct.
if a document was created directly from BSON data and has a Symbol or DBPointer type in it, we won't do anything. but if a value is read out from a doc, it will be upconverted to a different type.